src/main/kotlin/org/jitsi/jibri/util/JibriSubprocess.kt

```diff
@@ -55,31 +56,32 @@ class JibriSubprocess(
         }
     }
 
-    fun stop() {
-        logger.info("Stopping $name process")
-        processStatePublisher?.stop()
-        process?.apply {
-            try {
-                stop()
-                waitFor(10, TimeUnit.SECONDS)
-            } catch (t: Throwable) {
-                logger.error("Error trying to gracefully stop subprocess $name: $t")
-            }
-            if (isAlive) {
-                logger.error("$name didn't stop, killing $name")
-                destroyForcibly()
-                waitFor(10, TimeUnit.SECONDS)
-            }
-        }
+    private fun waitForProcessLoggerTaskToFinish(timeout: Duration) {
         try {
-            processLoggerTask?.get(10, TimeUnit.SECONDS)
+            processLoggerTask?.get(timeout.toMillis(), TimeUnit.MILLISECONDS)
         } catch (e: TimeoutException) {
             logger.error("Timed out waiting for process logger task to complete")
             processLoggerTask?.cancel(true)
         } catch (e: Exception) {
             logger.error("Exception while waiting for process logger task to complete", e)
             processLoggerTask?.cancel(true)
         }
+    }
+
+    fun stop() {
+        logger.info("Stopping $name process")
+        processStatePublisher?.stop()
+
+        process?.apply {
+            if (!stopAndWaitFor(Duration.ofSeconds(10))) {
+                logger.error("Error trying to gracefully stop $name, destroying forcibly")
+                if (!destroyForciblyAndWaitFor(Duration.ofSeconds(10))) {
+                    logger.error("Error trying to destroy $name forcibly")
```

Comment:
Follow up PR maybe to handle if forced stop failed.

Reply:
yeah i'm thinking we should go to an error state here.  will do that in another PR